tests/30rooms/05aliases.pl

```diff
@@ -147,3 +147,185 @@
          )->main::expect_http_403;
       })
    };
+
+
+test "Can delete canonical alias",
+   requires => [ local_user_fixture( with_events => 0 ), room_alias_fixture(),
+                 qw( can_create_room_alias )],
+
+   do => sub {
+      my ( $creator, $room_alias ) = @_;
+      my $server_name = $creator->http->server_name;
+      my $room_id;
+
+      matrix_create_and_join_room( [ $creator ] )
+      ->then( sub {
+         ( $room_id ) = @_;
+
+         do_request_json_for( $creator,
+            method => "PUT",
+            uri    => "/r0/directory/room/$room_alias",
+
+            content => { room_id => $room_id },
+         )
+      })->then( sub {
+         matrix_put_room_state( $creator, $room_id,
+            type => "m.room.canonical_alias",
+            content => { alias => $room_alias }
+         )
+      })->then( sub {
+         matrix_get_room_state( $creator, $room_id,
+            type      => "m.room.aliases",
+            state_key => $server_name,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( aliases ) );
+         assert_json_list( my $aliases = $body->{aliases} );
+
+         $room_alias ~~ @$aliases or die "Expected alias to be in list";
```

Comment:
Smartmatch is evil and should be taken out and shot.

Instead I'd suggest an `any` call:

    any { $_ eq $room_alias } @$aliases or
       die "Expected ..."